### PR TITLE
Client handle login error

### PR DESCRIPTION
**Pull request by ThomasNickerson**
This PR updates `get_auth_token` to not swallow errors passed back in the SignInUnion and instead raise an exception with error details. Also changed `create_transport` to `_create_transport` as there is no reason to access it from outside core.py.

### CODE REVIEW

File: clients/python/tracker_client/queries.py [hunk 1]
```diff
@@ -29,6 +29,10 @@
                 ... on AuthResult {
                     authToken
                 }
+                ... on SignInError{
```
Comment (nsdeschenes) [inline clients/python/tracker_client/queries.py:32]: This may be better off in a separate PR, however we may want to add support in for users who have tfa enabled.
```graphql
result {
  ... on TFASignInResult {
    sendMethod
    authenticateToken
  }
  ... on AuthResult {
    authToken
  }
  ... on SignInError {
    code
    description
  }
}
```

**Review by nsdeschenes — CHANGES_REQUESTED** [review]

File: clients/python/tracker_client/queries.py (hunk 1 above)
Comment (ThomasNickerson) [reply clients/python/tracker_client/queries.py:32]: I'll add it to the project board, I think that should definitely be added but in another PR.

**Review by tparrott-cse — APPROVED** [review]

**Review by nsdeschenes — APPROVED** [review]
